app/assets/javascripts/jquery_overrides.js

```diff
@@ -37,10 +37,20 @@ $.ajaxSetup({
     }),
     "text script": logError(function (text) {
       if (text.match(/^{/)) {
+        console.debug('json payload');
         return jQuery.jsonPayload(text, function (text) {
           return text;
         });
+
+      } else if (text.match(/<!DOCTYPE html>/)) { // plain HTML payload fallback
+        console.debug('html payload');
+        document.open();
+        document.write(text);
```

Comment:
This is a fall back. I was thinking about how to do this work w/o introducing some new bool query as we have `restful?` and checking on each link if the target supports partial updates.

I concluded that I don't want to do that as we would have already 4 variants: (restfull, non-restfull) x (supports_partial_updates, does_not_support....)

This way if we receive a full page in place where we expected a partial update we will use this fallback end everything will work.